dev/com.ibm.ws.kernel.boot.core/src/io/openliberty/checkpoint/spi/SnapshotResult.java

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright (c) 2021 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package io.openliberty.checkpoint.spi;
+
+import org.osgi.annotation.versioning.ProviderType;
+
+@ProviderType
+public class SnapshotResult {
+
+    public enum SnapshotResultType {
+        SUCCESS(1),
+        UNSUPPORTED_OPERATION(-1),
+        INVALID_ARGUMENTS(-2),
+        SYSTEM_CHECKPOINT_FAILURE(-3),
```

Comment:
I assume this is the type used when criu itself fails.  In that case we will probably have access to the actual error code from criu.  We may need to codes here.  One that is the liberty exit code and one that is the error code from criu dump result.